src/primitive_file_win.cc

```diff
@@ -309,88 +315,124 @@ PRIMITIVE(read) {
 
 PRIMITIVE(write) {
   ARGS(int, fd, Blob, bytes, int, from, int, to);
+  auto handle = reinterpret_cast<HANDLE>(fd);
   if (from > to || from < 0 || to > bytes.length()) FAIL(OUT_OF_BOUNDS);
   ssize_t current_offset = from;
   while (current_offset < to) {
-    ssize_t bytes_written = write(fd, bytes.address() + current_offset, to - current_offset);
-    if (bytes_written < 0) {
-      if (errno == EINTR) continue;
-      if (errno == EINVAL || errno == EBADF) FAIL(INVALID_ARGUMENT);
-      FAIL(ERROR);
-    }
+    DWORD bytes_written;
+    BOOL success = WriteFile(handle, bytes.address() + current_offset, to - current_offset, &bytes_written, NULL);
+    if (!success) WINDOWS_ERROR;
     current_offset += bytes_written;
   }
   return Smi::from(current_offset - from);
 }
 
 PRIMITIVE(close) {
   ARGS(int, fd);
+  HANDLE handle = reinterpret_cast<HANDLE>(fd);
   while (true) {
-    int result = close(fd);
-    if (result < 0) {
-      if (GetFileType(reinterpret_cast<HANDLE>(fd)) == FILE_TYPE_PIPE && errno == EBADF) {
+    int result = CloseHandle(handle);
+    if (result == 0) {
+      if (GetFileType(handle) == FILE_TYPE_PIPE && GetLastError() == ERROR_INVALID_HANDLE) {
         return process->null_object(); // Ignore already closed on PIPEs
       }
-      if (errno == EINTR) continue;
-      if (errno == EBADF) FAIL(ALREADY_CLOSED);
-      if (errno == ENOSPC) FAIL(QUOTA_EXCEEDED);
-      FAIL(ERROR);
+      WINDOWS_ERROR;
     }
     return process->null_object();
   }
 }
 
-Object* time_stamp(Process* process, time_t time) {
-  return Primitive::integer(time * 1000000000ll, process);
+int64 low_high_to_int64(DWORD high, DWORD low) {
+  return (((int64)high) << 32) + low;
```

Comment:
We don't allow C-style casts.
```suggestion
  int64 result = high;
  result <<= 32;
  result += low
  return result;
```